resources/prosody-plugins/util.lib.lua

```diff
@@ -416,25 +416,35 @@ function is_moderated(room_jid)
     return false;
 end
 
--- check if the room tenant starts with
--- vpaas-magic-cookie-
-function is_vpaas(room_jid)
-    local node, host = jid.split(room_jid);
+-- check if the room tenant starts with vpaas-magic-cookie-
+-- @param room the room to check
+function is_vpaas(room)
+    if not room then
+        return false;
+    end
+
+    -- stored check in room object if it exist
+    if room.is_vpaas ~= nil then
+        return room.is_vpaas;
+    end
+
+    room.is_vpaas = false;
+
+    local node, host = jid.split(room.jid);
```

Comment:
I just noticed other code checks if  room.jid is nil. Is that a possible case?

Reply:
I don't think so.